Dockerfile

```diff
@@ -1,7 +1,16 @@
-FROM golang:alpine3.11
+FROM golang:alpine3.11 as base
 
 WORKDIR /app
 COPY . /app
 
+ENV CGO_ENABLED=0
 RUN go build -o bin/naveed
-CMD ["/app/bin/naveed"]
+
+FROM busybox
```

Comment:
what's this used for?

Reply:
this is a "multistage" Docker image: its as if you have a two docker files in one but you can access the first docker image. I build the binary in t he `golang:alpine` image (which contains the go compiler, etc.) and copy the result in a new image that is based on `busybox`, a minimal Linux Distro. This way, the resulting image is much smaller (384mb vs. 12mb)